src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,84 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * Make this class immutable. See requirements in the README file
  */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        this.year = year;
+        this.color = color;
+        this.wheels = new ArrayList<>();
+        initWheels(wheels);
```

Comment:
```suggestion
        this.wheels = getWheelsCopy(wheels);
```
let's create `getWheelsCopy(wheels)` method that we will be able to reuse
this method should create a deep list copy - that is, create a new list and fill it with copies of all wheels